UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/FileShareServiceCache.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using UKHO.FileShareClient.Models;
+using UKHO.MaritimeSafetyInformation.Common.Configuration;
+using UKHO.MaritimeSafetyInformation.Common.Helpers;
+using UKHO.MaritimeSafetyInformation.Common.Logging;
+using UKHO.MaritimeSafetyInformation.Common.Models.AzureTableEntities;
+using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
+using UKHO.MaritimeSafetyInformation.Web.Services.Interfaces;
+
+namespace UKHO.MaritimeSafetyInformation.Web.Services
+{
+    public class FileShareServiceCache : IFileShareServiceCache
+    {
+        private readonly IAzureTableStorageClient _azureTableStorageClient;
+        private readonly IOptions<CacheConfiguration> _cacheConfiguration;
+        private readonly IAzureStorageService _azureStorageService;
+        private readonly ILogger<FileShareServiceCache> _logger;
+        private readonly IUserService _userService;
+
+        public FileShareServiceCache(IAzureTableStorageClient azureTableStorageClient, IOptions<CacheConfiguration> cacheConfiguration, IAzureStorageService azureStorageService, ILogger<FileShareServiceCache> logger, IUserService userService)
+        {
+            _azureTableStorageClient = azureTableStorageClient;
+            _cacheConfiguration = cacheConfiguration;
+            _azureStorageService = azureStorageService;
+            _logger = logger;
+            _userService = userService;
+        }
+
+        public async Task<BatchAttributesSearchModel> GetAllYearWeekFromCache(string rowKey, string correlationId)
+        {
+            string partitionKey = _userService.IsDistributorUser ? "Distributor" : "Public";
+            BatchAttributesSearchModel SearchResult = new();
```

Comment:
Camel case for local variables please.
```suggestion
            BatchAttributesSearchModel searchResult = new();
```

Reply:
done required changes